src/flutter/shell/platform/linux_embedded/window/elinux_window_wayland.cc

```diff
@@ -604,7 +606,10 @@ const wl_output_listener ELinuxWindowWayland::kWlOutputListener = {
           self->view_properties_.height = height;
 
           if (self->window_decorations_) {
-            self->window_decorations_->Resize(width, height);
+            int32_t width_dip = width / self->current_scale_;
+            int32_t height_dip = height / self->current_scale_;
```

Comment:
Need to divide by `self->current_scale_ is correct` here?

Reply:
Yes, according to my reading of the [spec](https://wayland.app/protocols/wayland#wl_output:event:mode) I think this is correct. The `width` and `height` here refer to the output *mode* which is in *physical* pixels while the resize method expects *logical* pixels. As an example, for a typical 4k HighDPI screen with an output scale of 2, the mode will be `3840 x 2160` but the logical size is `1920 x 1080`.

That being said, I'm not sure if resizing the window in response to the `wl_output.mode` event is the right thing to do. For one thing, this logic will break on devices with multiple outputs.

My impression is that it might be enough to only resize the window in response to [`xdg_toplevel.configure`](https://wayland.app/protocols/xdg-shell#xdg_toplevel:event:configure) events and get rid of this resizing logic from the [`wl_output.mode`](https://wayland.app/protocols/wayland#wl_output:event:mode) event. But I haven't tested this so I may be totally wrong here.